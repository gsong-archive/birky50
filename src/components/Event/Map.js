//@flow strict
import React from "react";

import { css } from "emotion";

import WindowSizeContext from "../../contexts/WindowSizeContext";
import { boxShadow1 } from "../../styles";

export default class Map extends React.Component<{}> {
  defaultWidth = 400;
  mapContainer = React.createRef();

  render = () => (
    <div ref={this.mapContainer}>
      <WindowSizeContext.Consumer>{this.renderMap}</WindowSizeContext.Consumer>
    </div>
  );

  renderMap = () => {
    const width = this.mapContainer.current
      ? this.mapContainer.current.clientWidth
      : this.defaultWidth;
    const height = getHeightFromWidth(width);

    return (
      <iframe
        src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2665.5868734347823!2d-114.24015338435312!3d48.07961667921895!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x5366535871876905%3A0x3088f2289bea6c!2sVista+Linda+Mexican+%26+Catering!5e0!3m2!1sen!2sus!4v1520397563655"
        title="map"
        height={height}
        allowFullScreen
        className={css`
          ${boxShadow1};
          width: 100%;
          border: 0;
        `}
      />
    );
  };
}

const getHeightFromWidth = width => {
  const max = 500;
  const min = 300;
  let height = Math.min(Math.floor((width / 4) * 3), max);
  height = Math.max(height, min);
  return height;
};
