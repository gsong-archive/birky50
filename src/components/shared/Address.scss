.address-link {
  padding-left: 1.5rem;
  text-indent: -1.5rem;
}
