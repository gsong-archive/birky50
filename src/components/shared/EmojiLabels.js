import React from "react";

const EmojiLabel = ({ emoji, ariaLabel, label }) => (
  <React.Fragment>
    <span role="img" aria-label={ariaLabel}>
      {emoji}
    </span>{" "}
    {label}
  </React.Fragment>
);

export const Airport = ({ label }) => (
  <EmojiLabel emoji="✈️" aria-label="airplane" label={label} />
);

export const Burger = ({ label }) => (
  <EmojiLabel emoji="🍔" aria-label="burger" label={label} />
);

export const Hotel = ({ label }) => (
  <EmojiLabel emoji="🏩" aria-label="hotel" label={label} />
);

export const PartyHat = ({ label }) => (
  <EmojiLabel emoji="🎉" aria-label="party hat" label={label} />
);

export const WomanBiking = ({ label }) => (
  <EmojiLabel emoji="🚴🏽‍♀️" aria-label="biking woman" label={label} />
);

export const WomanRaisingHand = ({ label }) => (
  <EmojiLabel emoji="🙋🏻‍♀️" aria-label="woman raising hand" label={label} />
);
