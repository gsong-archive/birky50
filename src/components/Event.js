import React from "react";

import facepaint from "facepaint";
import styled, { css } from "react-emotion";

import Address from "./shared/Address";
import PhoneNumber from "./shared/PhoneNumber";
import RSVPLink from "./shared/RSVPLink";
import SectionHeader from "./shared/SectionHeader";
import WindowSizeContext from "../contexts/WindowSizeContext";
import { Calendar } from "./shared/EmojiLabels";
import { Link } from "../styles/components";

import eventIcs from "../static/files/event.ics";

export default ({ LabelComponent, sectionLabel }) => {
  return (
    <React.Fragment>
      <SectionHeader>
        <LabelComponent label={sectionLabel} />
      </SectionHeader>

      <Event>
        <EventDetails>
          <h2>When?</h2>
          <TightP>Saturday, June 23, 2018</TightP>
          <TightP>Starts at 4:00pm, dinner buffet at 5:00pm</TightP>
          <Link href={eventIcs} download>
            <Calendar label="Add to calendar" />
          </Link>

          <h2
            className={css`
              margin-top: 1.5rem;
            `}
          >
            Where?
          </h2>
          <Link href="http://vistalindacatering.com/" rel="nofollow">
            Vista Linda Mexican & Catering
          </Link>
          <address>
            <Address
              address="240 Boon Rd, Somers, MT 59932"
              url="https://goo.gl/maps/THRyvih62562"
            />
            <PhoneNumber number="(406) 857-3158" />
          </address>

          <ul
            className={css`
              padding: 0;
            `}
          >
            <LI>Event will take place outdoors in a covered pavilion</LI>
            <LI>Dressy casual attire</LI>
          </ul>

          <RSVPLink width="100%" />
        </EventDetails>
        <EventMap>
          <Map />
        </EventMap>
      </Event>
    </React.Fragment>
  );
};

class Map extends React.Component {
  defaultWidth = 400;
  mapContainer = React.createRef();

  render = () => (
    <div ref={this.mapContainer}>
      <WindowSizeContext.Consumer>{this.renderMap}</WindowSizeContext.Consumer>
    </div>
  );

  renderMap = () => {
    const width = this.mapContainer.current
      ? this.mapContainer.current.clientWidth
      : this.defaultWidth;
    const height = getHeightFromWidth(width);

    return (
      <iframe
        src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2665.5868734347823!2d-114.24015338435312!3d48.07961667921895!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x5366535871876905%3A0x3088f2289bea6c!2sVista+Linda+Mexican+%26+Catering!5e0!3m2!1sen!2sus!4v1520397563655"
        title="map"
        width="100%"
        height={height}
        frameBorder="0"
        allowFullScreen
      />
    );
  };
}

const getHeightFromWidth = width => {
  const max = 500;
  const min = 300;
  let height = Math.min(Math.floor(width / 4 * 3), max);
  height = Math.max(height, min);
  return height;
};

const LI = styled("li")`
  margin-left: 1.5rem;
`;

const TightP = styled("p")`
  margin: 0.25rem 0;
`;

const eventItem = css`
  flex-grow: 0;
  flex-shrink: 0;
  padding: 0 1rem;
`;

const breakpoints = [900];
const mq = facepaint(breakpoints.map(bp => `@media (min-width: ${bp}px)`));

const Event = styled("div")(
  css`
    display: flex;
    margin: 0 -1rem;
  `,
  mq({
    flexWrap: ["wrap", "nowrap"]
  })
);

const EventDetails = styled("div")(
  eventItem,
  mq({
    flexBasis: ["100%", "calc(100% / 3)"]
  })
);

const EventMap = styled("div")(
  eventItem,
  mq({
    flexBasis: ["100%", "calc(100% / 3 * 2)"],
    marginTop: ["1.5rem", "0"]
  })
);
