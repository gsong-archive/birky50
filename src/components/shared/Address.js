//@flow strict
import React from "react";

import { Link } from "../../styles/components";
import { Location } from "./EmojiLabels";

type Props = {
  address: string,
  url: string,
};

export default ({ address, url }: Props) => (
  <div>
    <Location label={<Link href={url}>{address}</Link>} />
  </div>
);
