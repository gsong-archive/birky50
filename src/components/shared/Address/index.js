//@flow strict
import React from "react";

import { Link } from "../../../styles/components";
import { Location } from "../EmojiLabels";

type Props = {
  address: string,
  /** Link to map */
  url: string,
};

/**
 * Render an address set inside a link (preferably a map)
 */
export default ({ address, url }: Props) => (
  <div>
    <Location label={<Link href={url}>{address}</Link>} />
  </div>
);
