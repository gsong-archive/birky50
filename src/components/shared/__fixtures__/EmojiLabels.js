//@flow
import React from "react";

import { createFixture } from "react-cosmos-flow/fixture";

import { generate } from "testUtils";

import { Airplane, EmojiLabel } from "../EmojiLabels";
import { Link } from "../../../styles/components";

export default [
  createFixture({
    name: "Predefined EmojiLabel with text label",
    component: Airplane,
    props: { label: generate.words() },
  }),
  createFixture({
    name: "Predefined EmojiLabel with link label",
    description: "`label` prop is a React node.",
    component: Airplane,
    props: {
      label: <Link href={generate.url()}>{generate.words()}</Link>,
    },
  }),
  createFixture({
    name: "With text label",
    component: EmojiLabel,
    props: {
      emoji: "ðŸ˜œ",
      ariaLabel: "stuck out tongue winking eye",
      label: generate.words(),
    },
  }),
  createFixture({
    name: "With link label",
    description: "`label` prop is a React node.",
    component: EmojiLabel,
    props: {
      emoji: "ðŸ˜œ",
      ariaLabel: "stuck out tongue winking eye",
      label: <Link href={generate.url()}>{generate.words()}</Link>,
    },
  }),
];
