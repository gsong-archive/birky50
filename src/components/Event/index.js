//@flow
import React from "react";

import facepaint from "facepaint";
import styled, { css } from "react-emotion";

import Address from "../shared/Address";
import PhoneNumber from "../shared/PhoneNumber";
import RSVPLink from "../shared/RSVPLink";
import WindowSizeContext from "../../contexts/WindowSizeContext";
import { Calendar } from "../shared/EmojiLabels";
import { Link } from "../../styles/components";
import { boxShadow1 } from "../../styles";
import { calcSpacing, convertListToEm } from "../../styles/utils";
import { supportsGrid } from "../../styles/cssFeatures";
import { textColors } from "../../styles/variables";

import eventIcs from "../../static/files/event.ics";

export default () => {
  return (
    <Event>
      <EventDetails>
        <div>
          <EventDetailTitle>When?</EventDetailTitle>
          <EventDetailContent>
            <p
              className={css`
                font-weight: bold;
              `}
            >
              <time dateTime="2018-06-23 16:00">Saturday, June 23, 2018</time>
            </p>
            <p>Starts at 4:00pm, dinner buffet at 5:00pm</p>
            <Link
              href={eventIcs}
              className={css`
                display: inline-block;
              `}
              download
            >
              <Calendar label="Add to calendar" />
            </Link>
          </EventDetailContent>
        </div>

        <div>
          <EventDetailTitle className={whereHeader}>Where?</EventDetailTitle>
          <EventDetailContent>
            <Link
              href="http://vistalindacatering.com/"
              rel="nofollow"
              className={css`
                font-weight: bold;
              `}
            >
              Vista Linda Mexican & Catering
            </Link>
            <address>
              <Address
                address="240 Boon Rd, Somers, MT 59932"
                url="https://goo.gl/maps/THRyvih62562"
              />
              <PhoneNumber number="(406) 857-3158" />
            </address>
          </EventDetailContent>
        </div>

        <ul
          className={css`
            ${spanned};
            padding: 0;
            margin-top: 2.25em;
          `}
        >
          <LI>Event will take place outdoors in a covered pavilion</LI>
          <LI>Dressy casual attire</LI>
        </ul>

        <RSVPLink
          width="100%"
          className={css`
            ${spanned};
            margin-top: 0.375rem;
          `}
        />
      </EventDetails>
      <EventMap>
        <Map />
      </EventMap>
    </Event>
  );
};

class Map extends React.Component<{}> {
  defaultWidth = 400;
  mapContainer = React.createRef();

  render = () => (
    <div ref={this.mapContainer}>
      <WindowSizeContext.Consumer>{this.renderMap}</WindowSizeContext.Consumer>
    </div>
  );

  renderMap = () => {
    const width = this.mapContainer.current
      ? this.mapContainer.current.clientWidth
      : this.defaultWidth;
    const height = getHeightFromWidth(width);

    return (
      <iframe
        src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2665.5868734347823!2d-114.24015338435312!3d48.07961667921895!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x5366535871876905%3A0x3088f2289bea6c!2sVista+Linda+Mexican+%26+Catering!5e0!3m2!1sen!2sus!4v1520397563655"
        title="map"
        height={height}
        allowFullScreen
        className={css`
          ${boxShadow1};
          width: 100%;
          border: 0;
        `}
      />
    );
  };
}

const getHeightFromWidth = width => {
  const max = 500;
  const min = 300;
  let height = Math.min(Math.floor((width / 4) * 3), max);
  height = Math.max(height, min);
  return height;
};

const EventDetailTitle = styled.h2`
  font-size: 1rem;
  font-weight: normal;
  color: ${textColors.gray};
  margin: 0 0 0;
`;

const EventDetailContent = styled.div`
  a,
  p {
    display: block;
    margin: ${calcSpacing(1.5)}em 0 0;
  }
`;

const LI = styled.li`
  color: ${textColors.lightGray};
  margin-left: 1.5rem;
`;

const breakpoints1 = convertListToEm([1010]);
const mq1 = facepaint(breakpoints1.map(bp => `@media (min-width: ${bp}em)`));

const Event = styled.div(
  mq1({
    display: [null, "grid"],
    gridTemplateColumns: [null, "1fr 2fr"],
    gridColumnGap: [null, "1.125rem"],
  }),
  !supportsGrid &&
    mq1({
      display: [null, "flex"],
      justifyContent: [null, "space-between"],
      "> *:first-child": {
        flex: [null, `0 0 32%`],
      },
      "> *:last-child": {
        flex: [null, `0 0 65%`],
      },
    })
);

const EventMap = styled.div(
  mq1({
    marginTop: ["1.5rem", "0"],
  })
);

const spanned = css(mq1({ gridColumn: ["1 / 3"] }));

const breakpoints2 = convertListToEm([590, 1010]);
const mq2 = facepaint(breakpoints2.map(bp => `@media (min-width: ${bp}em)`));

const EventDetails = styled.div(
  mq2({
    display: [null, "grid", "block"],
    gridTemplateColumns: [null, "1fr 1fr"],
    gridColumnGap: [null, "10vw"],
  }),
  !supportsGrid &&
    mq2({
      display: [null, "flex", "block"],
      flexWrap: "wrap",
      "> *:first-child": {
        marginRight: [null, "10vw"],
      },
    })
);

const whereHeader = css(mq2({ marginTop: ["2.25em", 0, "2.25em"] }));
