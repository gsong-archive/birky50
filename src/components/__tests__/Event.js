import React from "react";

import { render } from "react-testing-library";

import { mockSupportsGrid } from "testUtils/mocks";

import Event from "../Event";
import { PartyHat } from "../shared/EmojiLabels";

mockSupportsGrid(true);

test("Event renders properly", () => {
  const { container } = render(
    <Event LabelComponent={PartyHat} sectionLabel="Celebration Details" />
  );

  expect(container).toMatchSnapshot();
});
