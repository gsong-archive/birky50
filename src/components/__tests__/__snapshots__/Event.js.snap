// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders properly 1`] = `
.emotion-12 {
  margin-top: 1.5rem;
}

@media (min-width:33.75em) {
  .emotion-12 {
    margin-top: 0;
  }
}

@media (min-width:56.25em) {
  .emotion-12 {
    margin-top: 1.5rem;
  }
}

.emotion-27 {
  grid-column: 1 / 3;
  padding: 0;
  margin-top: 1.5rem;
}

.emotion-2 {
  line-height: 1.1;
  margin-bottom: 1rem;
}

@media (max-width:21.875em) {
  .emotion-2 {
    font-size: 2.4rem;
  }
}

.emotion-1 {
  display: inline-block;
  margin-left: 1.4em;
  text-indent: -1.4em;
}

.emotion-0 {
  font-family: "Segoe UI Emoji";
}

@media (min-width:56.25em) {
  .emotion-35 {
    display: grid;
    grid-template-columns: 1fr 2fr;
    grid-column-gap: 1rem;
  }
}

@media (min-width:33.75em) {
  .emotion-30 {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 3rem;
  }
}

@media (min-width:56.25em) {
  .emotion-30 {
    display: block;
  }
}

.emotion-4 {
  margin: 0.25rem 0;
}

.emotion-10 {
  color: maroon;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-23 {
  margin-left: 1.5rem;
}

.emotion-28 {
  border-radius: 0.25rem;
  display: inline-block;
  font-size: 1rem;
  text-align: center;
  vertical-align: middle;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.5rem 0.75rem;
  font-size: 1.25rem;
  padding: 0.75rem 1rem;
  border-radius: 0.3rem;
  color: white;
  font-weight: bold;
  background-color: rgb(225,18,130);
  width: 100%;
  grid-column: 1 / 3;
  margin-top: 0.5rem;
}

.emotion-33 {
  margin-top: 1.5rem;
}

@media (min-width:56.25em) {
  .emotion-33 {
    margin-top: 0px;
  }
}

.emotion-32 {
  width: 100%;
  border: 0;
}

<div>
  <h1
    id="uniqueId-description"
    class="emotion-2 emotion-3"
  >
    <span
      class="emotion-1"
    >
      <span
        class="emotion-0"
        role="img"
        aria-label="party hat"
      >
        üéâ
      </span>
      Celebration Details
    </span>
  </h1>
  <div
    class="emotion-35 emotion-36"
  >
    <div
      class="emotion-30 emotion-31"
    >
      <div>
        <h2>
          When?
        </h2>
        <p
          class="emotion-4 emotion-5"
        >
          <time
            datetime="2018-06-23 16:00"
          >
            Saturday, June 23, 2018
          </time>
        </p>
        <p
          class="emotion-4 emotion-5"
        >
          Starts at 4:00pm, dinner buffet at 5:00pm
        </p>
        <a
          href="event.ics"
          download=""
          class="emotion-10 emotion-11"
        >
          <span
            class="emotion-1"
          >
            <span
              class="emotion-0"
              role="img"
              aria-label="calendar"
            >
              üìÜ
            </span>
            Add to calendar
          </span>
        </a>
      </div>
      <div>
        <h2
          class="emotion-12"
        >
          Where?
        </h2>
        <a
          href="http://vistalindacatering.com/"
          rel="nofollow"
          class="emotion-10 emotion-11"
        >
          Vista Linda Mexican & Catering
        </a>
        <address>
          <div>
            <span
              class="emotion-1"
            >
              <span
                class="emotion-0"
                role="img"
                aria-label="round pushpin"
              >
                üìç
              </span>
              <a
                href="https://goo.gl/maps/THRyvih62562"
                class="emotion-10 emotion-11"
              >
                240 Boon Rd, Somers, MT 59932
              </a>
            </span>
          </div>
          <div>
            <span
              class="emotion-1"
            >
              <span
                class="emotion-0"
                role="img"
                aria-label="phone receiver"
              >
                üìû
              </span>
              <a
                href="tel:(406) 857-3158"
                class="emotion-10 emotion-11"
              >
                (406) 857-3158
              </a>
            </span>
          </div>
        </address>
      </div>
      <ul
        class="emotion-27"
      >
        <li
          class="emotion-23 emotion-24"
        >
          Event will take place outdoors in a covered pavilion
        </li>
        <li
          class="emotion-23 emotion-24"
        >
          Dressy casual attire
        </li>
      </ul>
      <a
        href="https://goo.gl/forms/SKdVBcUt7LUQmP1K2"
        width="100%"
        class="emotion-28 emotion-29"
      >
        RSVP
      </a>
    </div>
    <div
      class="emotion-33 emotion-34"
    >
      <div>
        <iframe
          src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2665.5868734347823!2d-114.24015338435312!3d48.07961667921895!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x5366535871876905%3A0x3088f2289bea6c!2sVista+Linda+Mexican+%26+Catering!5e0!3m2!1sen!2sus!4v1520397563655"
          title="map"
          height="300"
          allowfullscreen=""
          class="emotion-32"
        />
      </div>
    </div>
  </div>
</div>
`;
